File: references/2-示例代码/Java/1-统一支付下单/UnifiedPayExample.java
package com.joinpay.sdk.example;

import com.alibaba.fastjson.JSONObject;
import com.joinpay.sdk.JoinPayClient;

/**
 * 1. 统一支付下单示例
 *
 * 演示如何调用汇聚支付统一支付接口，支持微信/支付宝/银联各支付方式。
 * 关键参数: orderNo(唯一), amount(元), productName, frpCode, notifyUrl, tradeMerchantNo
 */
public class UnifiedPayExample {

    public static void main(String[] args) throws Exception {
        // ===== 初始化客户端 =====
        JoinPayClient client = new JoinPayClient(
            "https://trade.joinpay.com",   // 生产环境地址
            "your_merchant_no",            // 商户号
            "your_merchant_key_32chars",   // MD5密钥
            null,                           // RSA私钥(MD5模式不需要)
            "MD5"                           // 签名方式: MD5 或 RSA
        );

        // ===== 构建请求 =====
        JoinPayClient.UnifiedPayRequest request = new JoinPayClient.UnifiedPayRequest();
        request.orderNo        = "ORDER" + System.currentTimeMillis();  // 唯一订单号
        request.amount         = "0.01";                               // 金额(元)
        request.productName    = "测试商品";                            // 商品名称
        request.frpCode        = "WEIXIN_GZH";                         // 交易类型
        request.notifyUrl      = "https://your-domain.com/notify";     // 回调地址
        request.tradeMerchantNo = "your_trade_merchant_no";             // 报备商户号

        // --- 渠道特定参数（根据frpCode选择性传入） ---
        // 微信公众号(GZH): 必传 openId + appId
        request.openId = "oXXXXXXXXXXXXXXXXX";  // 用户OpenId
        request.appId  = "wxXXXXXXXXXXXXXXXX";  // 公众号AppId

        // 微信被扫(CARD): 必传 authCode + terminalNo + terminalIp
        // request.authCode   = "134567890123456789";
        // request.terminalNo = "TERMINAL001";
        // request.terminalIp = "192.168.1.100";

        // 支付宝H5: 可选 transactionModel (MODEL1/MODEL2/MODEL3)
        // request.transactionModel = "MODEL1";

        // ===== 发起下单 =====
        JSONObject response = client.unifiedPay(request);

        // ===== 处理结果 =====
        String ra_Code = response.getString("ra_Code");
        if ("100".equals(ra_Code)) {
            // ✅ 下单成功
            String trxNo   = response.getString("r7_TrxNo");       // 平台交易流水号
            String result  = response.getString("rc_Result");      // 二维码URL/HTML/跳转链接等
            String pic     = response.getString("rd_Pic");          // base64二维码图片(如q4=1)

            System.out.println("✅ 下单成功!");
            System.out.println("交易流水号: " + trxNo);
            System.out.println("结果数据:  " + result);

            // 根据不同支付方式处理rc_Result:
            if ("WEIXIN_NATIVE".equals(request.frpCode)) {
                // 主扫: result是二维码URL，展示给用户扫
                System.out.println("请展示二维码给用户扫描: " + result);
            } else if ("WEIXIN_GZH".equals(request.frpCode) || "WEIXIN_XCX".equals(request.frpCode)) {
                // 公众号/小程序: result是调起支付的JSON
                System.out.println("调起支付参数: " + result);
            } else if ("ALIPAY_H5".equals(request.frpCode)) {
                // H5: result可能是HTML或重定向URL
                System.out.println("H5支付页面: " + result);
            }
        } else {
            // ❌ 下单失败
            String rb_CodeMsg = response.getString("rb_CodeMsg");
            System.out.println("❌ 下单失败! code=" + ra_Code + ", msg=" + rb_CodeMsg);
            System.out.println("排查建议: 查看[📄排障手册]中的响应码速查表");
        }
    }
}

File: references/2-示例代码/Python/1-统一支付下单/unified_pay_example.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
1. 统一支付下单示例

演示如何调用汇聚支付统一支付接口。
关键参数: orderNo(唯一), amount(元), productName, frpCode, notifyUrl, tradeMerchantNo
"""

import sys
import os
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from SDK工具类.joinpay_client import JoinPayClient


def main():
    # ===== 初始化客户端 =====
    client = JoinPayClient(
        base_url="https://trade.joinpay.com",
        merchant_no="your_merchant_no",
        merchant_key="your_merchant_key_32chars",
        rsa_private_key="",  # RSA模式时填写
        sign_type="MD5",    # MD5 或 RSA
        version="2.6",
    )

    # ===== 构建请求 =====
    response = client.unified_pay(
        order_no=f"ORDER{__import__('time').time()}",  # 唯一订单号
        amount="0.01",                               # 金额(元)
        product_name="测试商品",                        # 商品名称
        frp_code="WEIXIN_GZH",                         # 交易类型
        notify_url="https://your-domain.com/notify",     # 回调地址
        trade_merchant_no="your_trade_merchant_no",     # 报备商户号
        
        # --- 渠道特定参数（根据frpCode选择性传入）---
        open_id="oXXXXXXXXXXXXXXXXX",   # 微信公众号必传
        app_id="wxXXXXXXXXXXXXXXXX",   # 微信AppId
    )

    # ===== 处理结果 =====
    ra_code = response.get("ra_Code", "")
    if ra_code == "100":
        # ✅ 下单成功
        trx_no = response.get("r7_TrxNo", "")       # 平台交易流水号
        result = response.get("rc_Result", "")      # 二维码URL/HTML/跳转链接等
        pic = response.get("rd_Pic", "")            # base64二维码图片

        print(f"✅ 下单成功!")
        print(f"交易流水号: {trx_no}")
        print(f"结果数据:  {result}")

        # 根据不同支付方式处理 rc_Result:
        if "WEIXIN_NATIVE" == "WEIXIN_GZH":  # 实际用变量
            pass  # result是二维码URL，展示给用户扫
    else:
        # ❌ 下单失败
        rb_msg = response.get("rb_CodeMsg", "")
        print(f"❌ 下单失败! code={ra_code}, msg={rb_msg}")
        print("排查建议: 查看[📄排障手册]中的响应码速查表")


if __name__ == "__main__":
    main()
